<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>832</width>
    <height>601</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Do an AI ca nhan</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow{
	background-color: #AAAAAA;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="lblStartState">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>60</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>Trang thai bat dau:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnSolve">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>140</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">/*QPushButton{
	background-color: #EEE2DC;
	border-radius: 10px;
}*/
QPushButton {
    background-color: #007BFF;
    color: white;
    border-radius: 8px;
    padding: 6px 12px;
    font-size: 14px;
    font-weight: bold;
}

QPushButton:hover {
    background-color: #0056b3;
}

</string>
    </property>
    <property name="text">
     <string>Giai</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblTitle">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>20</y>
      <width>691</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #27A4F2;
}</string>
    </property>
    <property name="text">
     <string>AND OR SEARCH</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell1">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell4">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell5">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell6">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell7">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell8">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell9">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell3_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>340</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell7_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>450</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell8_3">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>450</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell9_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>450</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell5_3">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>400</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell4_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>400</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell1_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>340</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell2_3">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>340</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell6_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>400</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="lblResult">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>300</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>KQ:</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblStep">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>520</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}
</string>
    </property>
    <property name="text">
     <string>Buoc:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="txtStep">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>520</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: transparent;
	color: black
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="textInteractionFlags">
     <set>Qt::NoTextInteraction</set>
    </property>
   </widget>
   <widget class="QLabel" name="lblTotalStep">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>200</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>So buoc:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="txtTotalStep">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>200</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: transparent;
	color: black
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="textInteractionFlags">
     <set>Qt::NoTextInteraction</set>
    </property>
   </widget>
   <widget class="QLabel" name="lblSolveTime">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>240</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>Thoi gian:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="txtSolveTime">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>240</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: transparent;
	color: black
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="textInteractionFlags">
     <set>Qt::NoTextInteraction</set>
    </property>
   </widget>
   <widget class="QPushButton" name="btnWriteToFile">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>220</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">/*QPushButton{
	background-color: #EEE2DC;
	border-radius: 10px;
}*/
QPushButton#writeToFile {
    background-color: #28A745;
}

QPushButton#writeToFile:hover {
    background-color: #218838;
}</string>
    </property>
    <property name="text">
     <string> </string>
    </property>
   </widget>
   <widget class="QLabel" name="lblNote">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>280</y>
      <width>781</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string> </string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRandomInput">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>140</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">/*QPushButton{
	background-color: #EEE2DC;
	border-radius: 10px;
}*/
QPushButton#writeToFile {
    background-color: #28A745;
}

QPushButton#writeToFile:hover {
    background-color: #218838;
}</string>
    </property>
    <property name="text">
     <string>Random input</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblStartState_3">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>261</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>Toc do (s/step):</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="txtSolveSpeedPerStep">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>90</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="txt_plan_tree">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>330</y>
      <width>571</width>
      <height>251</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #FFFFFF;
	color: black
}</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="lblResult_2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>300</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>Cay:</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblStartState_2">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>60</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	color: #FF652F;
}</string>
    </property>
    <property name="text">
     <string>Trang thai ket thuc:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell9_end">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell2_end">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell3_end">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell6_end">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell1_end">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>100</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell4_end">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell8_end">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell7_end">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>210</y>
      <width>51</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="cell5_end">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>160</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPlainTextEdit{
	background-color: #CFEBFC;
	color: black
}</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>832</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
